File: January/Java/Range Sum of BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  public int rangeSumBST(TreeNode node, int low, int high) {
    if (node == null) return 0;
    
    return (node.val >= low && node.val <= high ? node.val : 0) +
      (low < node.val ? rangeSumBST(node.left, low, high) : 0) +
      (high > node.val ? rangeSumBST(node.right, low, high) : 0);
  }
}

File: January/cpp/Range Sum of BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int sum =0;
public:

    int rangeSumBST(TreeNode* root, int low, int high) {
    if (root == NULL){
        return 0;
    }
    else{
        rangeSumBST(root->left,low,high);
        int s = root->val;

        if(s>=low && s<=high){
            sum+=s;
        }
        rangeSumBST(root->right,low,high);
    }
    return sum;
    }
};
